modules/portaudio/portaudio.c

```diff
@@ -223,10 +223,11 @@ static int src_alloc(struct ausrc_st **stp, const struct ausrc *as,
 	if (!stp || !as || !prm)
 		return EINVAL;
 
-	if (str_isset(device))
-		dev_index = atoi(device);
-	else
-		dev_index = Pa_GetDefaultInputDevice();
+	dev = !str_cmp(device, "default")
```

Comment:
Could you please use `str_casecmp` here to make it more robust ?

Reply:
You can update the PR if you want.

The important part is that the device string should handle values like integer,string,default and empty.